docs/differences-from-v1.md

```diff
@@ -0,0 +1,69 @@
+
+In 1.x, the CLI was configured by "modules", which were entrypoints to user programs that the CLI should analyze.
+
+In 2.x, we've renamed "modules" to "analysis targets", and simplified how analysis targets are defined.
+
+We renamed analysis targets to avoid confusing naming conflicts with existing language-specific "modules". Analysis targets are no longer defined by multiple, confusingly similar fields (e.g. "Dir" vs. "Path"), and are now simple `(type, target)` tuples.
+
+### Automatic analysis target discovery
+
+FOSSA CLI 2.x now does automatic analysis target discovery when you run `fossa analyze`.
+
+This means that `fossa init` is no longer required. `fossa init` is now a no-op that emits a warning, and will be removed in a future release.
+
+### Automatic analysis target strategy selection
+
+In 1.x, modules were manually configured with "strategies" that specified _how_ a module should be analyzed. While `fossa init` attempted to choose the best strategy, manual intervention was sometimes required depending on the project's setup.
+
+In 2.x, the CLI now automatically selects the optimal strategy for analysis targets given the current environment (e.g. whether a build tool is available). This is possible because discovery and analysis are now one step, so we can check the suitability of analysis strategies while discovering targets.
+
+### Improved correctness
+
+Analysis strategies in CLI 2.x have substantive improvements in correctness and reliability across all language integrations. 2.x has much stronger compile-time correctness guarantees in its parsers, and uses build tool plugins instead of output parsing for target types where output parsing was brittle.
+
+### Improved debug logging
+
+CLI 2.x has much better debug logging, including a new feature called "replay logging" which allows developers to perfectly reproduce a bug report given a replay log. This is made possible by stronger compile-time guarantees that ensure all effects that occur during analysis are logged for replay.
+
+## How to upgrade to FOSSA 2.x
+
+### Remove calls to `fossa init`
+
+Since analysis targets are now automatically discovered during analysis, `fossa init` is no longer a valid command. `fossa init` is currently a no-op that emits a warning. It may be removed in a future release.
+
+### Migrate your configuration file
+
+We've made major breaking changes in the configuration file format for CLI 2.x to improve clarity. You'll need to migrate your 1.x configuration files to the new 2.x format for their configurations to apply.
+
+Configuration files for 1.x will be ignored when running 2.x. We determine whether a configuration file is for 1.x by examining its `version` field. Configuration file `version`s `1` and `2` are for 1.x.
+
+<!-- TODO: Write up configuration file field docs -->
+
+For documentation on the new configuration file format, see [here](https://github.com/fossas/spectrometer/pull/220).
+
+### Migrate "archive upload" targets
```

Comment:
It might be helpful to drop a link to the CLI v1 documentation in this section. https://github.com/fossas/fossa-cli/blob/master/docs/integrations/archive.md#archive. People know "archive upload" as a lot of different things like "raw target" or "license scan".